pkg/kubectl/cmd/portforward.go

```diff
@@ -76,11 +75,21 @@ func (*defaultPortForwarder) ForwardPorts(req *client.Request, config *client.Co
 
 func RunPortForward(f *cmdutil.Factory, cmd *cobra.Command, args []string, fw portForwarder) error {
 	podName := cmdutil.GetFlagString(cmd, "pod")
-	if len(podName) == 0 {
-		return cmdutil.UsageError(cmd, "POD_NAME is required for port-forward")
+	if len(podName) == 0 && len(args) == 0 {
```

Comment:
Doesn't it make sense to just move this check after the if/else below?  That way you're only testing `len(podName)`.